chain/indexer/src/streamer/types.rs

```diff
@@ -17,17 +17,40 @@ pub struct IndexerChunkView {
     pub header: views::ChunkHeaderView,
     pub transactions: Vec<IndexerTransactionWithOutcome>,
     pub receipts: Vec<views::ReceiptView>,
-    pub receipt_execution_outcomes: Vec<IndexerExecutionOutcomeWithReceipt>,
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct IndexerTransactionWithOutcome {
     pub transaction: views::SignedTransactionView,
-    pub outcome: IndexerExecutionOutcomeWithReceipt,
+    pub outcome: IndexerExecutionOutcomeWithOptionalReceipt,
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct IndexerExecutionOutcomeWithReceipt {
+pub struct IndexerExecutionOutcomeWithOptionalReceipt {
     pub execution_outcome: views::ExecutionOutcomeWithIdView,
     pub receipt: Option<views::ReceiptView>,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct IndexerExecutionOutcomeWithReceipt {
+    pub execution_outcome: views::ExecutionOutcomeWithIdView,
+    pub receipt: views::ReceiptView,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct IndexerShard {
+    pub shard_id: types::ShardId,
+    pub chunk: Option<IndexerChunkView>,
+    pub receipt_execution_outcomes: Vec<IndexerExecutionOutcomeWithReceipt>,
+}
+
+impl From<IndexerExecutionOutcomeWithOptionalReceipt> for IndexerExecutionOutcomeWithReceipt {
+    fn from(outcome_with_optional_receipt: IndexerExecutionOutcomeWithOptionalReceipt) -> Self {
+        Self {
+            execution_outcome: outcome_with_optional_receipt.execution_outcome,
+            receipt: outcome_with_optional_receipt
+                .receipt
+                .expect("`receipt` must be present at this moment"),
+        }
+    }
+}
```

Comment:
I am not 100% against `From` implementation here, but it might be easier to reason about if we in fact inline this transformation where we need it, since "receipt must be present at **this** moment" is not super accurate without the context of "this moment" given that `From` does not imply that information.

Reply:
Agree, I had some doubts, you've just approved my theory about implicit way